projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-order-approval.ts

```diff
@@ -0,0 +1,133 @@
+import * as sampleData from '../../sample-data/b2b-order-approval';
+import {
+  approvalOrderDetail,
+  approvalOrderList,
+  approvedOrderDetails,
+  pendingOrder,
+  rejectedOrderDetails,
+} from '../../sample-data/b2b-order-approval';
+import { verifyTabbingOrder } from '../accessibility/tabbing-order';
+import { TabElement } from '../accessibility/tabbing-order.model';
+import { visitHomePage, waitForPage } from '../checkout-flow';
+
+export function visitOrderApprovalListPage() {
+  const alias = waitForPage(
+    '/my-account/approval-dashboard',
+    'approvalListPage'
+  );
+
+  cy.visit(`/my-account/approval-dashboard`);
+  cy.wait(`@${alias}`).its('status').should('eq', 200);
+}
+
+export function visitOrderApprovalDetailPage() {
+  const alias = waitForPage(
+    `/my-account/approval/${sampleData.approvalOrderDetail.code}`,
+    'approvalDetailPage'
+  );
+
+  cy.visit(`/my-account/approval/${sampleData.approvalOrderDetail.code}`);
+  cy.wait(`@${alias}`).its('status').should('eq', 200);
+}
+
+export function loginB2bUser() {
+  cy.requireLoggedIn(sampleData.b2bUserAccount);
+  visitHomePage();
+}
+
+export function loginB2bApprover() {
+  cy.requireLoggedIn(sampleData.b2bApproverAccount);
+  visitHomePage();
```

Comment:
I think you don't need to visit the homepage here. This takes about 3 seconds.

Reply:
Done.